src/ragger/conftest/base_conftest.py

```diff
@@ -0,0 +1,170 @@
+    for file_name, lib_name in configuration.OPTIONNAL_CONFIGURATION["SIDELOADED_APPS"].items():
+        lib_path = app_path_from_app_name(configuration.REQUIRED_CONFIGURATION["APPS_DIRECTORY"],
+                                          file_name, firmware.device)
+        speculos_args.append(f"-l{lib_name}:{lib_path}")
+
+    return (app_path, {"args": speculos_args})
+
+
+# Depending on the "--backend" option value, a different backend is
+# instantiated, and the tests will either run on Speculos or on a physical
+# device depending on the backend
+def create_backend(backend_name: str, firmware: Firmware, display: bool,
+                   log_apdu_file: Optional[Path]):
+    if backend_name.lower() == "ledgercomm":
+        return LedgerCommBackend(firmware=firmware, interface="hid", log_apdu_file=log_apdu_file)
+    elif backend_name.lower() == "ledgerwallet":
+        return LedgerWalletBackend(firmware=firmware, log_apdu_file=log_apdu_file)
+    elif backend_name.lower() == "speculos":
+        app_path, speculos_args = prepare_speculos_args(firmware, display)
+        return SpeculosBackend(app_path, firmware=firmware, log_apdu_file=log_apdu_file, **speculos_args)
+    else:
+        raise ValueError(f"Backend '{backend_name}' is unknown. Valid backends are: {BACKENDS}")
+
+
+# Backend scope can be configured by the user
+@pytest.fixture(scope=configuration.OPTIONNAL_CONFIGURATION["BACKEND_SCOPE"])
+def backend(backend_name, firmware, display, log_apdu_file):
+    with create_backend(backend_name, firmware, display, log_apdu_file) as b:
+        yield b
+
+
+@pytest.fixture
+def navigator(backend, firmware, golden_run):
+    if firmware.device.startswith("nano"):
+        return NanoNavigator(backend, firmware, golden_run)
+    else:
+        raise ValueError(f"Device '{firmware.device}' is unsupported.")
+
+
+@pytest.fixture
+def use_only_on_backend(request, backend_name):
+    if request.node.get_closest_marker('use_on_backend'):
+        current_backend = request.node.get_closest_marker('use_on_backend').args[0]
+        if current_backend != backend_name:
+            pytest.skip(f'skipped on this backend: "{current_backend}"')
+
+
+def pytest_configure(config):
+    config.addinivalue_line(
+        "markers",
+        "use_only_on_backend(backend): skip test if not on the specified backend",
+    )
+
+
+def print_full_conf():
+    print("Running Ragger with the following configuration:")
+    for key, value in configuration.REQUIRED_CONFIGURATION.items():
+        print(f"    {key} = '{value}'")
+    for key, value in configuration.OPTIONNAL_CONFIGURATION.items():
+        print(f"    {key} = '{value}'")
+
+
+def assert_full_conf():
+    missing = []
+    for key, value in configuration.REQUIRED_CONFIGURATION.items():
+        if value == "":
+            missing.append(key)
+    if missing:
+        raise ValueError(f"Missing required configuration parameters: '{missing}")
+
+
+# RUN ON IMPORT
+print_full_conf()
```

Comment:
This is a bit verbose? Or not that much?

Reply:
I agree, it is a bit much. Maybe using a proper `logger.debug` instead of `print` would avoid flooding the stdout all the time.

Reply:
Done